### PR TITLE
Renamed loading prop on button base element

### PR DESCRIPTION
**Pull request by woutsluiter**
### This PR:

removes the `loading` prop on the button base DOM element. The passing of this prop to the dom element is unnecessary and causes warnings in the latest `styled-components` version:

```
Warning: Received `false` for a non-boolean attribute `loading`.
If you want to write it to the DOM, pass a string instead: loading="false" or loading={value.toString()}.
```

**Breaking changes** 🔥
- none

**Backwards compatible additions** ✨
- none

**Bugfixes/Changed internals** 🎈
- Removed `loading` prop to button base DOM element

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable).
- [x] Appropriate tests have been added for my functionality (check if not applicable).
- [x] A designer has seen and approved my changes (tag `@LuukHorsmans` or `@RianneSchaekens` for a design review when applicable).
- [x] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers).
- [x] Appropriate documentation has been written (check if not applicable).

### CODE REVIEW

**Comment by woutsluiter** [comment]
hmm wait I think the styling now no longer works 🧐. But the non-HTMLprops should filtered out somehow. Back to the drawing board.

**Comment by CarloPalinckx** [comment]
Works on your deployment 🤷‍♂

File: yarn.lock [hunk 1]
```diff
@@ -26,6 +26,13 @@
   dependencies:
```
Comment (CarloPalinckx) [inline yarn.lock]: Could you revert these changes?

**Comment by woutsluiter** [comment]
> Works on your deployment 🤷‍♂

loading specific styling from the base component doesn't, like the cursor should be `default` instead of `pointer`

File: yarn.lock (hunk 1 above)
Comment (woutsluiter) [reply yarn.lock]: this is an update of `styled-component` and all its dependencies

File: src/components/Button/base/index.tsx [hunk 2]
```diff
@@ -41,15 +41,14 @@ const ButtonBase: FunctionComponent<PropsType> = (props): JSX.Element => {
...
-            {...props}
             title={props.title}
             className={props.className}
             onClick={clickAction}
             type="button"
             disabled={props.disabled}
             id={props.id}
-            loading={props.loading}
             data-testid={props['data-testid']}
+            data-loading={props.loading ? 'true' : 'false'}
```
Comment (CarloPalinckx) [inline src/components/Button/base/index.tsx]: this still gets set on the html element, you are better of just naming the prop "isLoading" to avoid these issues

Comment (CarloPalinckx) [reply src/components/Button/base/index.tsx]: > this still gets set on the html element, you are better of just naming the prop "isLoading" to avoid these issues
